usaspending_api/disaster/v2/views/agency/count.py

```diff
@@ -17,18 +19,28 @@ class AgencyCountViewSet(CountBase):
 
     @cache_response()
     def post(self, request: Request) -> Response:
-        filters = [
-            Q(treasury_account__funding_toptier_agency=OuterRef("pk")),
-            self.is_in_provided_def_codes(),
-            self.all_closed_defc_submissions,
-            self.is_non_zero_object_class_cpe(),
-        ]
-        count = (
-            ToptierAgency.objects.annotate(
-                include=Exists(FinancialAccountsByProgramActivityObjectClass.objects.filter(*filters).values("pk"))
+        if self.award_type_codes:
+            filters = [
+                self.is_in_provided_def_codes(),
+                self.all_closed_defc_submissions,
+                self.is_non_zero_award_cpe(),
+            ]
+            count = FinancialAccountsByAwards.objects.filter(*filters)
+            count = count.values("award_id").filter(award__type__in=self.filters.get("award_type_codes"))
+            count = count.aggregate(count=Count("award__funding_agency__toptier_agency", distinct=True))["count"]
+        else:
+            filters = [
+                Q(treasury_account__funding_toptier_agency=OuterRef("pk")),
+                self.is_in_provided_def_codes(),
+                self.all_closed_defc_submissions,
+                self.is_non_zero_object_class_cpe(),
```

Comment:
I recommend changing the name from `is_non_zero_object_class_cpe` to `is_non_zero_total_spending` ... where "total" is to indicate the `spending_type` field in play here (one of `total`, `awards`, `loans`, from the api contract and general domain language being used around the different type of disaster spending).

The trouble with the name as it is, is two-fold:
1. It doesn't follow suit with the existing `FinancialAccountsByProgramActivityObjectClass` model it implies reference to, because it only mentions `object_class` and not program activity
2. With just object class there (and no doc strings for the method!) it could easily be implied that this is a filter that has something to do with the object class dimension and not with spending amounts.

Reply:
I used the name `is_non_zero_object_class_cpe` to juxtapose it to `is_non_zero_award_cpe`, naming each vaguely after the table involved. I'm all for a better naming convention, particularly if it's more based on business logic, but I think `is_non_zero_total_spending` vs `is_non_zero_award_cpe` will still be confusing. 
  Possibly `is_non_zero_total_award_spending` vs `is_non_zero_total_object_program_spending`? That seems to bring us back to the original problem though.

Reply:
On second looks, I think your suggestion makes a lot of sense. I've renamed to `is_non_zero_total_spending` and `is_non_zero_award_spending`, with hopes that a future refactor will bring `is_non_zero_loan_spending` alongside them one day.